_posts/2021-09-30-spring-i18n.md

```diff
@@ -0,0 +1,379 @@
+---
+title: "How to Internationalize a Spring Boot Application"
+categories: [spring-boot]
+date: 2021-09-30 06:00:00 +1000
+modified: 2021-09-30 06:00:00 +1000
+author: pratikdas
+excerpt: "Internationalization is the process of making an application adaptable to multiple languages and regions without major changes in the source code. In this article, we will understand the concepts of internationalization, and illustrate how to internationalize a Spring Boot application."
+image:
+  auto: 0074-stack
+---
+
+Internationalization is the process of making an application adaptable to multiple languages and regions without major changes in the source code.
+
+In this article, we will understand the concepts of internationalization, and illustrate how to internationalize a Spring Boot application.
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/spring-boot/spring-boot-i18n" %}
+
+## Internationalization (i18n) vs Localization (l10n)
+
+Internationalization is a mechanism to create multilingual software that can be adapted to different languages and regions.
+
+An internationalized application has the following characteristics:
+
+- The application can be adapted to run in multiple regions by adding region or language-specific configuration data.
+- Text elements like information messages and the user interface labels, are stored outside the source code and retrieved at runtime.
+- Addition of support for new languages does not require code changes.
+- Culturally-dependent data like dates and currencies are displayed in formats of the end user's region and language.
+
+Internationalization is also abbreviated as i18n because there is a total of 18 characters between the first letter 'i' and the last letter 'n'.
+
+The following figure is an example of Amazon's e-commerce site supporting internationalization:
+
+![i18n example](/assets/img/posts/spring-boot-i18n/i18nexample.png)
```

Comment:
Can you split this image in two (one for German, one for French)? It's hard to discern where one starts and the other ends.

Reply:
cc @pratikdas